src/sys/time.rs

```diff
@@ -58,6 +60,20 @@ const TS_MAX_SECONDS: i64 = ::std::isize::MAX as i64;
 
 const TS_MIN_SECONDS: i64 = -TS_MAX_SECONDS;
 
+impl From<Duration> for TimeSpec {
```

Comment:
I think there should be some doctests for `TimeSpec` demonstrating these conversions as they can be hard to discover unless we put them in examples.

Reply:
Done.